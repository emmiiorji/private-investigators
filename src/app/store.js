import { configureStore } from '@reduxjs/toolkit';
import investigatorsReducer from './redux/investigator/investigatorSlice';

const store = configureStore({
  reducer: {
    investigators: investigatorsReducer,
  },
});

export default store;
