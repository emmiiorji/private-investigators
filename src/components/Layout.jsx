import { Outlet, NavLink } from 'react-router-dom';
import '../assets/stylesheets/nav.css';
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  FaTwitter, FaFacebookF, FaVimeoV, FaPinterestP,
} from 'react-icons/fa';
import { TiSocialInstagram } from 'react-icons/ti';
import { Button } from 'react-bootstrap';
import { signOutUser } from '../redux/CurrentUser/CurrentUserSlice';

const Layout = () => {
  const dispatch = useDispatch();

  const handleLogout = () => {
    dispatch(signOutUser());
  }

  const currentUser = useSelector((state) => state.userReducer);
  return (
    <div className="layout">
      <nav className="d-flex flex-column">
        <ul className="d-flex flex-column">
          <li>
            <NavLink to="/app/investigators">Home</NavLink>
          </li>
          <li>
            <NavLink to="/app/appointments">My Appointments</NavLink>
          </li>
          <li>
            <NavLink to="/app/investigators/:id/reserve">Book an Appointment</NavLink>
          </li>
          <Button onClick={ handleLogout } variant="danger">Logout</Button>
          <hr />
          { currentUser.admin
          && (
            <>
              <li>
                <NavLink to="/app/add-investigator">Add an Investigator</NavLink>
              </li>
              <li>
                <NavLink to="/app/delete-investigator/:id">Delete Investigator</NavLink>
              </li>
            </>
          )}
        </ul>
        <ul className="social-icons">
          <li><a href="https://www.twitter.com" aria-label="Twitter"><FaTwitter /></a></li>
          <li><a href="https://www.facebook.com" aria-label="Facebook"><FaFacebookF /></a></li>
          <li><a href="https://www.instagram.com" aria-label="Instagram"><TiSocialInstagram /></a></li>
          <li><a href="https://www.vimeo.com" aria-label="Vimeo"><FaVimeoV /></a></li>
          <li><a href="https://www.pinterest.com" aria-label="Pinterest"><FaPinterestP /></a></li>
        </ul>
        <footer>
          <p>
            &copy; 2023 Mátyás Gombos, Waris Haleem, Chris Clothier,
            Felipe Oyarzo and Emmanuel Orji-Ihuoma
          </p>
        </footer>
      </nav>
      <Outlet />
    </div>
  );
};

export default Layout;
