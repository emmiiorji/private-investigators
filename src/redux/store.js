import { configureStore } from '@reduxjs/toolkit';
// import userReducer from './CurrentUser/CurrentUserSlice';
import investigatorsReducer from './investigator/investigatorSlice';
import appointmentsReducer from './Appointments/appointmentsSlice';
import userReducer from './CurrentUser/CurrentUserSlice';

const store = configureStore({
  reducer: {
    investigators: investigatorsReducer,
    appointments: appointmentsReducer,
    userReducer,
  },
});

export default store;
