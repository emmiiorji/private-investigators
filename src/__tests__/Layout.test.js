import '@testing-library/jest-dom';
import { BrowserRouter as Router, NavLink } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Layout from '../components/Layout';

describe('Testing the Layout Component', () => {
  let root;
  beforeAll(async () => {
    root = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Layout />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('Component should render correctly', () => {
    expect(Layout).toBeTruthy();
  });

  it('It should match its snapshot', () => {
    expect(root).toMatchSnapshot();
  });

  it('It renders navigation links', () => {
    waitFor(() => expect(NavLink.getByText('Home')).toBeInTheDocument());
    waitFor(() => expect(NavLink.getByText('My Appointments')).toBeInTheDocument());
    waitFor(() => expect(NavLink.getByText('Book an Appointment')).toBeInTheDocument());
  });
});
