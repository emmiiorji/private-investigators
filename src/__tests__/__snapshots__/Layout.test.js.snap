// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout Component It should match its snapshot 1`] = `
<div
  className="layout"
>
  <nav>
    <ul>
      <li>
        <a
          className=""
          href="/investigators"
          onClick={[Function]}
        >
          Investigators
        </a>
      </li>
      <li>
        <a
          className=""
          href="/add-investigator"
          onClick={[Function]}
        >
          Add an Investigator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/appointments"
          onClick={[Function]}
        >
          My Appointments
        </a>
      </li>
      <li>
        <a
          className=""
          href="/add-appointment"
          onClick={[Function]}
        >
          Book an Appointment
        </a>
      </li>
      <li>
        <a
          className=""
          href="/delete-investigator/:id"
          onClick={[Function]}
        >
          Delete Investigator
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
